### ISSUE TITLE
Line number reported multiple times when multiple nodes of structured operators fail

### ISSUE THREAD

**Issue by VeraBE** [Azure/template-analyzer#117]
When analyzing: [azure-quickstart-templates\kubernetes-on-ubuntu-vmss\azuredeploy.json](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/kubernetes-on-ubuntu-vmss/azuredeploy.json)

Errors reported with repeated line numbers;
```
        Kubernetes_Must_Restrict_Network_Access: Authorized IP ranges should be defined on Kubernetes Services
                Result: Failed
                Line: 149
                Line: 149
        Kubernetes_Must_Use_RBAC: Role-Based Access Control (RBAC) should be used on Kubernetes Services
                Result: Failed
                Line: 149
                Line: 149
```

The rules use allOf and anyOf.  In each rule, both nodes of the structured operators fail, and the line number is reported for each node failure.

**Comment by JohnathonMohr** [comment]
This was mostly fixed as part of #227.  Other scenarios where this might still happen are tracked in #228.
